zml/aio/torch.zig

```diff
@@ -314,79 +314,76 @@ pub const PickleData = struct {
+                                        try self.parseValue(allocator, store, new_prefix, val);
+                                    },
                                 }
-                            }.call;
-                        }
-                    }.call;
-
-                    if (seq[1].len > 0 and switch (seq[1][0]) {
-                        inline .int, .bool, .float => |_, tag| utils.allTrue(seq[1][1..], elemCheck(tag)),
-                        else => false,
-                    }) {
-                        const out: []u8 = switch (seq[1][0]) {
-                            .int => blk: {
-                                const d = try allocator.alloc(i64, seq[1].len);
-                                for (seq[1], 0..) |item, i| {
-                                    d[i] = item.int;
-                                }
-                                break :blk std.mem.sliceAsBytes(d);
-                            },
-                            .float => blk: {
-                                const d = try allocator.alloc(f64, seq[1].len);
-                                for (seq[1], 0..) |item, i| {
-                                    d[i] = item.float;
-                                }
-                                break :blk std.mem.sliceAsBytes(d);
-                            },
-                            else => blk: {
-                                const d = try allocator.alloc(bool, seq[1].len);
-                                for (seq[1], 0..) |item, i| {
-                                    d[i] = item.bool;
-                                }
-                                break :blk std.mem.sliceAsBytes(d);
-                            },
-                        };
-                        const key = try allocator.dupe(u8, prefix.items);
-                        const d = try store._metadata.getOrPut(allocator, key);
-                        if (d.found_existing) {
-                            log.warn("Duplicate key: {s}", .{prefix.items});
-                            allocator.free(key);
-                            allocator.free(out);
-                        } else d.value_ptr.* = @unionInit(zml.aio.Value, "array", .{ .item_type = switch (seq[1][0]) {
-                            .int => .int64,
-                            .float => .float64,
-                            .string => .string,
-                            else => .boolval,
-                        }, .data = out });
-                    } else {
-                        for (seq[1], 0..) |item, i| {
-                            var new_prefix = prefix;
-                            if (v.isPrimitive()) {
-                                if (prefix.items.len > 0) {
-                                    new_prefix.appendAssumeCapacity('.');
+                            }
+                            const used_tag = switch (tag) {
```

Comment:
this feels like an another argument to have less Value types.

Reply:
I think we want to support a diverse set of metadata types though, so not sure sacrificing there makes sense. I've updated the code so the enum values used for the tagged union align with the enum values used in `zml.aio.Value.Slice.ItemType` (e.g. `.bool` becomes `.boolval`, etc.).